sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java

```diff
@@ -496,6 +511,11 @@ public ServiceBusSenderAsyncClient buildAsyncClient() {
             final MessagingEntityType entityType = validateEntityPaths(logger, connectionStringEntityName, topicName,
                 queueName);
 
+            if (!CoreUtils.isNullOrEmpty(viaQueueName) && !CoreUtils.isNullOrEmpty(topicName)) {
```

Comment:
You should use `entityType` instead of topicName for your comparison

Reply:
Also add documentation for this runtime error. It should be an IllegalStateException.